src/System.Numerics.Vectors/tests/GenericVectorTests.cs

```diff
@@ -2610,6 +2611,76 @@ public void NarrowDouble()
         #endregion Narrow / Widen
 
         #region Helper Methods
+        private static void AssertEqual<T>(T expected, T actual, string operation, int precision = -1) where T : IEquatable<T>
+        {
+            if (typeof(T) == typeof(float))
+            {
+                if (!IsDiffTolerable((float)(object)expected, (float)(object)actual, precision))
+                {
+                    throw new XunitException($"AssertEqual failed for operation {operation}. Expected: {expected}, Actual: {actual}.");
```

Comment:
Made these changes. Thanks for pointing out this digit padding syntax, never used it before.